tests/suites/test_suite_psa_crypto.function

```diff
@@ -1990,32 +1999,85 @@ exit:
 /* BEGIN_CASE */
 void hash_bad_order( )
 {
+    psa_algorithm_t alg = PSA_ALG_SHA_256;
     unsigned char input[] = "";
     /* SHA-256 hash of an empty string */
-    unsigned char hash[] = {
+    const unsigned char valid_hash[] = {
         0xe3, 0xb0, 0xc4, 0x42, 0x98, 0xfc, 0x1c, 0x14, 0x9a, 0xfb, 0xf4, 0xc8,
         0x99, 0x6f, 0xb9, 0x24, 0x27, 0xae, 0x41, 0xe4, 0x64, 0x9b, 0x93, 0x4c,
         0xa4, 0x95, 0x99, 0x1b, 0x78, 0x52, 0xb8, 0x55 };
+    unsigned char hash[sizeof(valid_hash)] = { 0 };
     size_t hash_len;
     psa_hash_operation_t operation = PSA_HASH_OPERATION_INIT;
 
     PSA_ASSERT( psa_crypto_init( ) );
 
-    /* psa_hash_update without calling psa_hash_setup beforehand */
-    memset( &operation, 0, sizeof( operation ) );
+    /* Call setup twice in a row. */
```

Comment:
This is getting to be a long series of ad hoc tests. Have we reached the point where they would be easier to write and easier to read if we made the code data-driven? That would make it easier to add other sequences.
```
for(c in string) {
    switch(c) {
    case 'a': ret = abort();
    case 'u': ret = update();
    ...
    }
    TEST_EQUAL(ret, c is last ? PSA_SUCCESS : expected_last_status);
}
abort();
setup(); update(); finish(); // test that the operation object is usable after abort
```

Reply:
I did have this thought as well, but decided it wasn't worth the time yet.